homeassistant/components/alarm_control_panel/egardia.py

```diff
@@ -71,40 +75,57 @@ def setup_platform(hass, config, add_devices, discovery_info=None):
     rs_enabled = config.get(CONF_REPORT_SERVER_ENABLED)
     rs_port = config.get(CONF_REPORT_SERVER_PORT)
     rs_codes = config.get(CONF_REPORT_SERVER_CODES)
+    version = config.get(CONF_VERSION)
 
     try:
         egardiasystem = egardiadevice.EgardiaDevice(
-            host, port, username, password, '')
+            host, port, username, password, '', version)
     except requests.exceptions.RequestException:
         raise exc.PlatformNotReady()
     except egardiadevice.UnauthorizedError:
         _LOGGER.error("Unable to authorize. Wrong password or username")
         return False
 
+    if rs_enabled:
+        # Set up the egardia server
+        _LOGGER.info("Setting up EgardiaServer")
+        try:
+            egardiaserver = egardiaserver.EgardiaServer('', rs_port)
```

Comment:
You're overwriting the module. Please use a different name for the variable.

Reply:
Also make sure you set the variable to `None` if `rs_enabled` is `False`.

Reply:
I also think we should store the server in `hass.data`, since there must only be one server, but a platform can be setup multiple times. So first check if there is a server in `hass.data['egardia']` and only make a new server if there isn't a server already.